### PR TITLE
Make model writers handle generic types

### PR DESCRIPTION
**Pull request by Gasparila**
ActiveRecord is very lenient when it comes to setting a column value.
Any Numeric subclass can be used (including Rational and BigDecimal)
when setting a value. Also remove DateTime as it is a subclass of Date
and therefore superfluous

### CODE REVIEW

**Comment by mahmoudian1** [comment]
# [Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=h1) Report
> Merging [#292](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=desc) into [master](https://codecov.io/gh/chanzuckerberg/sorbet-rails/commit/19c2d4a372b03bc339b973c530de6da01f8b990e&el=desc) will **decrease** coverage by `1.52%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #292      +/-   ##
==========================================
- Coverage   96.79%   95.26%   -1.53%     
==========================================
  Files         142      102      -40     
  Lines        2526     2028     -498     
==========================================
- Hits         2445     1932     -513     
- Misses         81       96      +15     
```


| [Impacted Files](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [...rbet-rails/model\_plugins/active\_storage\_methods.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9wbHVnaW5zL2FjdGl2ZV9zdG9yYWdlX21ldGhvZHMucmI=) | `50.00% <0.00%> (-50.00%)` | :arrow_down: |
| [lib/sorbet-rails/utils.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy91dGlscy5yYg==) | `90.90% <0.00%> (-9.10%)` | :arrow_down: |
| [spec/rails\_helper.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9yYWlsc19oZWxwZXIucmI=) | `28.26% <0.00%> (-6.53%)` | :arrow_down: |
| [lib/sorbet-rails/model\_rbi\_formatter.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9yYmlfZm9ybWF0dGVyLnJi) | `90.00% <0.00%> (-0.63%)` | :arrow_down: |
| [...b/sorbet-rails/model\_plugins/active\_record\_enum.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9wbHVnaW5zL2FjdGl2ZV9yZWNvcmRfZW51bS5yYg==) | `89.18% <0.00%> (-0.56%)` | :arrow_down: |
| [...rbet-rails/rails\_mixins/active\_record\_overrides.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9yYWlsc19taXhpbnMvYWN0aXZlX3JlY29yZF9vdmVycmlkZXMucmI=) | `95.18% <0.00%> (-0.17%)` | :arrow_down: |
| [lib/sorbet-rails/sorbet\_utils.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9zb3JiZXRfdXRpbHMucmI=) | `97.56% <0.00%> (-0.17%)` | :arrow_down: |
| [...bet-rails/model\_plugins/active\_record\_attribute.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9wbHVnaW5zL2FjdGl2ZV9yZWNvcmRfYXR0cmlidXRlLnJi) | `97.77% <0.00%> (+1.18%)` | :arrow_up: |
| [.../sorbet-rails/model\_plugins/active\_record\_assoc.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9wbHVnaW5zL2FjdGl2ZV9yZWNvcmRfYXNzb2MucmI=) | `98.18% <0.00%> (-0.10%)` | :arrow_down: |
| [lib/sorbet-rails/model\_utils.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF91dGlscy5yYg==) | `97.36% <0.00%> (-0.07%)` | :arrow_down: |
| ... and [50 more](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=footer). Last update [19c2d4a...bdbc5ff](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by mahmoudian1** [comment]
# [Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=h1) Report
> Merging [#292](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=desc) into [master](https://codecov.io/gh/chanzuckerberg/sorbet-rails/commit/19c2d4a372b03bc339b973c530de6da01f8b990e&el=desc) will **decrease** coverage by `1.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #292      +/-   ##
==========================================
- Coverage   96.79%   95.76%   -1.03%     
==========================================
  Files         142      126      -16     
  Lines        2526     2316     -210     
==========================================
- Hits         2445     2218     -227     
- Misses         81       98      +17     
```


| [Impacted Files](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [spec/support/v6.0/config/environment.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvY29uZmlnL2Vudmlyb25tZW50LnJi) | `50.00% <0.00%> (-50.00%)` | :arrow_down: |
| [...rbet-rails/model\_plugins/active\_storage\_methods.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy9tb2RlbF9wbHVnaW5zL2FjdGl2ZV9zdG9yYWdlX21ldGhvZHMucmI=) | `50.00% <0.00%> (-50.00%)` | :arrow_down: |
| [spec/support/v6.0/config/application.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvY29uZmlnL2FwcGxpY2F0aW9uLnJi) | `75.00% <0.00%> (-25.00%)` | :arrow_down: |
| [lib/sorbet-rails/utils.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-bGliL3NvcmJldC1yYWlscy91dGlscy5yYg==) | `90.90% <0.00%> (-9.10%)` | :arrow_down: |
| [spec/support/v5.0/sorbet\_test\_cases.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y1LjAvc29yYmV0X3Rlc3RfY2FzZXMucmI=) | `100.00% <0.00%> (ø)` | |
| [spec/support/v5.1/sorbet\_test\_cases.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y1LjEvc29yYmV0X3Rlc3RfY2FzZXMucmI=) | `100.00% <0.00%> (ø)` | |
| [spec/support/v6.0/app/helpers/baz\_helper.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvYXBwL2hlbHBlcnMvYmF6X2hlbHBlci5yYg==) | | |
| [spec/support/v6.0/app/helpers/bar\_helper.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvYXBwL2hlbHBlcnMvYmFyX2hlbHBlci5yYg==) | | |
| [....0/config/initializers/filter\_parameter\_logging.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvY29uZmlnL2luaXRpYWxpemVycy9maWx0ZXJfcGFyYW1ldGVyX2xvZ2dpbmcucmI=) | | |
| [...ort/v6.0/app/mailers/hogwarts\_acceptance\_mailer.rb](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree#diff-c3BlYy9zdXBwb3J0L3Y2LjAvYXBwL21haWxlcnMvaG9nd2FydHNfYWNjZXB0YW5jZV9tYWlsZXIucmI=) | | |
| ... and [12 more](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292/diff?src=pr&el=tree-more) | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=footer). Last update [19c2d4a...bdbc5ff](https://codecov.io/gh/chanzuckerberg/sorbet-rails/pull/292?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by manhhung741** [comment]
Currently 

> This branch is out-of-date with the base branch

Can you rebase your branch on top of the current master? I'll merge after it's updated

File: spec/test_data/v5.0/expected_robe.rbi [hunk 1]
```diff
@@ -13,7 +13,7 @@ module Robe::GeneratedAttributeMethods
   sig { returns(Integer) }
   def id; end
 
-  sig { params(value: T.any(Integer, Float, ActiveSupport::Duration)).void }
+  sig { params(value: T.any(Numeric, ActiveSupport::Duration)).void }
```
[the remark this answers is not in the document]
Comment (Gasparila) [reply spec/test_data/v5.0/expected_robe.rbi]: Yep. Basically I want to be able to set `container.weight = 2` for a BigDecimal column.

**Comment by Gasparila** [comment]
@manhhung741 It looks like the 2.4 tests are failing due to errors inside of Parlor. I'm guessing that's expected? If not I can look into it

**Comment by manhhung741** [comment]
Don’t worry, I’ve flagged the issue to Parlour’s owner. It’s likely will be
a fix in sorbet master. (We test against sorbet master). When it’s fixed
I’ll rerun the checks for you :-)

**Pull request merged**